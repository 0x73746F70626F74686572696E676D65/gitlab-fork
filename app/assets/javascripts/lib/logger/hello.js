const HANDSHAKE = String.fromCodePoint(0x1f91d);
const MAG = String.fromCodePoint(0x1f50e);

export const logHello = () => {
  // eslint-disable-next-line no-console
  console.log(
    `%cWelcome to GitLab!%c

Does this page need fixes or improvements? Open an issue or contribute a merge request to help make GitLab more lovable. At GitLab, everyone can contribute!

${HANDSHAKE} Contribute to GitLab: https://about.gitlab.com/community/contribute/
${MAG} Create a new GitLab issue: https://gitlab.com/gitlab-org/gitlab/-/issues/new`,
    `padding-top: 0.5em; font-size: 2em;`,
    'padding-bottom: 0.5em;',
  );
};
