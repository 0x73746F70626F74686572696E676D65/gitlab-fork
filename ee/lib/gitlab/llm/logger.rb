# frozen_string_literal: true

module Gitlab
  module Llm
    class Logger < ::Gitlab::JsonLogger
      def self.file_name_noext
        'llm'
      end

      def self.log_level
        Gitlab::Utils.to_boolean(ENV['LLM_DEBUG']) ? ::Logger::DEBUG : ::Logger::INFO
      end

      def info_or_debug(user, message)
        if Feature.enabled?(:expanded_ai_logging, user)
          info(message)
        else
          debug(message)
        end
      end
    end
  end
end
