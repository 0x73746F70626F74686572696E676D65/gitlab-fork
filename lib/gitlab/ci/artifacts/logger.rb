# frozen_string_literal: true

module Gitlab
  module Ci
    module Artifacts
      module Logger
        def log_artifacts_filesize(artifact_file)
          return if artifact_file.nil?

          unless artifact_file.is_a?(::Ci::Artifactable)
            raise ArgumentError, "unknown artifact file class `#{artifact_file.class}`"
          end

          ::Gitlab::ApplicationContext.push(artifact: artifact_file)
        end

        def log_artifacts_context(job)
          ::Gitlab::ApplicationContext.push(
            namespace: job&.project&.namespace,
            project: job&.project,
            job: job
          )
        end

        def log_build_dependencies(size:, count: 0)
          ::Gitlab::ApplicationContext.push(
            artifacts_dependencies_size: size,
            artifacts_dependencies_count: count
          )
        end
      end
    end
  end
end
