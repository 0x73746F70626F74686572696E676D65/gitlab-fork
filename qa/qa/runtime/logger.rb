# frozen_string_literal: true

require 'logger'
require 'forwardable'

module QA
  module Runtime
    module Logger
      extend SingleForwardable

      LEVEL_COLORS = {
        "DEBUG" => :magenta,
        "INFO" => :green,
        "WARN" => :yellow,
        "ERROR" => :indianred,
        "FATAL" => :red
      }.freeze

      def_delegators :logger, :debug, :info, :warn, :error, :fatal, :unknown

      singleton_class.module_eval do
        attr_writer :logger

        def logger
          Rainbow.enabled = Runtime::Env.colorized_logs?

          @logger ||= ::Logger.new(Runtime::Env.log_destination).tap do |logger|
            logger.level = Runtime::Env.debug? ? ::Logger::DEBUG : ::Logger::ERROR

            logger.formatter = proc do |severity, datetime, progname, msg|
              date_format = datetime.strftime("%Y-%m-%d %H:%M:%S")
              msg_prefix = "[date=#{date_format} from=QA Tests] #{severity.ljust(5)} -- "

              Rainbow(msg_prefix).send(LEVEL_COLORS.fetch(severity, :yellow)) + "#{msg}\n"
            end
          end
        end
      end
    end
  end
end
