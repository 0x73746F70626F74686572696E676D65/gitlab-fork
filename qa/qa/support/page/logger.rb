# frozen_string_literal: true

module QA
  module Support
    module Page
      module Logger
        def refresh
          super

          log("refreshed #{current_url}")
        end

        def wait(max: 60, time: 0.1, reload: true)
          log("with wait: max #{max}; time #{time}; reload #{reload}")

          super

          log("end wait")
        end

        def scroll_to(selector, text: nil)
          log("scrolling to :#{selector}")

          super
        end

        def asset_exists?(url)
          exists = super

          log("asset_exists? #{url} returned #{exists}")

          exists
        end

        def find_element(name)
          element = super

          log("found :#{name}")

          element
        end

        def all_elements(name)
          elements = super

          log("found all :#{name}")

          elements
        end

        def click_element(name)
          super

          log("clicked :#{name}")
        end

        def fill_element(name, content)
          super

          content = '*****' if name.to_s.include? "password"

          log(%Q(filled :#{name} with "#{content}"))
        end

        def has_element?(name)
          found = super

          log("has_element? :#{name} returned #{found}")

          found
        end

        def within_element(name)
          log("within element :#{name}")

          super

          log("end within element :#{name}")
        end

        private

        def log(msg)
          puts "debug: #{msg}"
        end
      end
    end
  end
end
