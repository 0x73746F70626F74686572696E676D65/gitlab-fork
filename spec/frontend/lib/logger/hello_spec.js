import { logHello } from '~/lib/logger/hello';

describe('~/lib/logger/hello', () => {
  let consoleLogSpy;

  beforeEach(() => {
    // We don't `mockImplementation` so we can validate there's no errors thrown
    consoleLogSpy = jest.spyOn(console, 'log');
  });

  describe('logHello', () => {
    it('console logs a friendly hello message', () => {
      expect(consoleLogSpy).not.toHaveBeenCalled();

      logHello();

      expect(consoleLogSpy.mock.calls).toMatchSnapshot();
    });
  });
});
