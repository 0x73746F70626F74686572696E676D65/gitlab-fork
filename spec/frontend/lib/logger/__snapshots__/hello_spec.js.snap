// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`~/lib/logger/hello logHello console logs a friendly hello message 1`] = `
Array [
  Array [
    "%cWelcome to GitLab!%c

Does this page need fixes or improvements? Open an issue or contribute a merge request to help make GitLab more lovable. At GitLab, everyone can contribute!

ü§ù Contribute to GitLab: https://about.gitlab.com/community/contribute/
üîé Create a new GitLab issue: https://gitlab.com/gitlab-org/gitlab/-/issues/new",
    "padding-top: 0.5em; font-size: 2em;",
    "padding-bottom: 0.5em;",
  ],
]
`;
